projects/client/RabbitMQ.Client/src/client/content/StreamWireFormatting.cs

```diff
@@ -195,71 +196,73 @@ public static object ReadNonnullObject(string target, NetworkBinaryReader reader
         public static object ReadObject(NetworkBinaryReader reader)
         {
             int typeTag = reader.ReadByte();
-            switch (typeTag)
+            if (typeTag == -1)
             {
-                case -1:
-                    throw new EndOfStreamException("End of StreamMessage reached");
+                throw new EndOfStreamException("End of StreamMessage reached");
+            }
 
-                case (int) StreamWireFormattingTag.Bool:
-                {
-                    byte value = reader.ReadByte();
-                    switch (value)
+            switch ((StreamWireFormattingTag)typeTag)
+            {
+                case StreamWireFormattingTag.Bool:
                     {
-                        case 0x00:
-                            return false;
-                        case 0x01:
-                            return true;
-                        default:
+                        byte value = reader.ReadByte();
+                        switch (value)
                         {
-                            string message =
-                                string.Format("Invalid boolean value in StreamMessage: {0}", value);
-                            throw new ProtocolViolationException(message);
+                            case 0x00:
+                                return false;
+                            case 0x01:
+                                return true;
+                            default:
+                                {
+                                    string message =
+                                        string.Format("Invalid boolean value in StreamMessage: {0}", value);
```

Comment:
for better performance string.Concat can be used 
`string.Concat("Invalid boolean value in StreamMessage: ", value.ToString())`.

See that I also suggest `value.ToString()` to reduce boxing of the value type.